File: Gemini_Pro/Java/Facade/Round1/Car_code1.java
public class Car_code1 {

    static class Engine {
        private boolean started;

        public void start() {
            started = true;
            System.out.println("สตาร์ทเครื่องยนต์");
        }

        public void stop() {
            started = false;
            System.out.println("ดับเครื่องยนต์");
        }

        public boolean isStarted() {
            return started;
        }
    }

    static class Lights {
        private boolean on;

        public void turnOn() {
            on = true;
            System.out.println("เปิดไฟหน้า");
        }

        public void turnOff() {
            on = false;
            System.out.println("ปิดไฟหน้า");
        }

        public boolean isOn() {
            return on;
        }
    }

    static class SteeringWheel {
        private String direction = "center";

        public void turnLeft() {
            direction = "left";
            System.out.println("เลี้ยวซ้าย");
        }

        public void turnRight() {
            direction = "right";
            System.out.println("เลี้ยวขวา");
        }

        public String getDirection() {
            return direction;
        }
    }

    static class Brakes {
        private boolean applied;

        public void apply() {
            applied = true;
            System.out.println("เบรก");
        }

        public void release() {
            applied = false;
            System.out.println("ปลดเบรก");
        }

        public boolean isApplied() {
            return applied;
        }
    }

    public static class CarFacade {
        private Engine engine = new Engine();
        private Lights lights = new Lights();
        private SteeringWheel steering = new SteeringWheel();
        private Brakes brakes = new Brakes();

        public void startCar() {
            System.out.println("Starting the car...");
            engine.start();
            lights.turnOn();
            System.out.println("Car is ready to go!");
        }

        public void stopCar() {
            System.out.println("Stopping the car...");
            engine.stop();
            lights.turnOff();
            System.out.println("Car has been stopped.");
        }

        public void turnLeft() {
            System.out.println("Executing left turn...");
            steering.turnLeft();
        }

        public void turnRight() {
            System.out.println("Executing right turn...");
            steering.turnRight();
        }

        public void applyBrakes() {
            System.out.println("Applying brakes...");
            brakes.apply();
        }

        public void releaseBrakes() {
            System.out.println("Releasing brakes...");
            brakes.release();
        }

        public Engine getEngine() {
            return engine;
        }

        public Lights getLights() {
            return lights;
        }

        public SteeringWheel getSteering() {
            return steering;
        }

        public Brakes getBrakes() {
            return brakes;
        }
    }

    public static void main(String[] args) {
        Car_code.CarFacade car = new Car_code.CarFacade();
        car.startCar();
        car.turnLeft();
        car.applyBrakes();
        car.releaseBrakes();
        car.turnRight();
        car.stopCar();
    }
}

File: Gemini_Pro/Python/Facade/Round1/Car_code1.py
class Engine:
    """Subsystem: เครื่องยนต์"""
    def start(self):
        print("สตาร์ทเครื่องยนต์")

    def stop(self):
        print("ดับเครื่องยนต์")


class Lights:
    """Subsystem: ไฟรถ"""
    def turn_on(self):
        print("เปิดไฟหน้า")

    def turn_off(self):
        print("ปิดไฟหน้า")


class SteeringWheel:
    """Subsystem: พวงมาลัย"""
    def turn_left(self):
        print("เลี้ยวซ้าย")

    def turn_right(self):
        print("เลี้ยวขวา")


class Brakes:
    """Subsystem: เบรก"""
    def apply(self):
        print("เบรก")

    def release(self):
        print("ปลดเบรก")


class CarFacade:
    def __init__(self):
        self.engine = Engine()
        self.lights = Lights()
        self.steering = SteeringWheel()
        self.brakes = Brakes()

    def start_car(self):
        print("Starting the car...")
        self.engine.start()
        self.lights.turn_on()
        print("Car is ready to go!")

    def stop_car(self):
        print("Stopping the car...")
        self.engine.stop()
        self.lights.turn_off()
        print("Car has been stopped.")

    def turn_left(self):
        print("Executing left turn...")
        self.steering.turn_left()

    def turn_right(self):
        print("Executing right turn...")
        self.steering.turn_right()

    def apply_brakes(self):
        print("Applying brakes...")
        self.brakes.apply()

    def release_brakes(self):
        print("Releasing brakes...")
        self.brakes.release()

# ใช้งาน CarFacade
car = CarFacade()# เริ่มต้นการใช้งานรถยนต์
car.start_car()
car.turn_left() # เลี้ยวซ้าย
car.apply_brakes() # เบรก
car.release_brakes() # ปลดเบรกและเลี้ยวขวา
car.turn_right()
car.stop_car() # ดับเครื่องยนต์และปิดไฟ